### ISSUE TITLE
CS0282 misjudgment

### ISSUE THREAD

**Issue by ufcpp** [dotnet/roslyn#23668]
**Version Used**: 

VS 15.5.0

**Steps to Reproduce**:

```cs
partial struct X // Warning CS0282
{
    // The only field of X is a backing field of A.
    public int A { get; set; }
}

partial struct X : I
{
    // This partial definition has no field.
    int I.A { get => A; set => A = value; }
}

interface I
{
    int A { get; set; }
}
```

**Expected Behavior**:

No warning.

**Actual Behavior**:

![image](https://user-images.githubusercontent.com/3760955/33758621-45853520-dc42-11e7-9e12-9cfd49ad0958.png)

FYI, this misjudgment occurs only when `I.A` has expression-bodied get/set.

```cs
partial struct X // No Warning
{
    public int A { get; set; }
}

partial struct X : I
{
    int I.A { get { return A; } set { A = value; } } // {}-bodied
}
```

**Comment by jcouv** [comment]
Thanks for reporting this. Fix is being reviewed.

**Comment by jcouv** [comment]
Fixed for 15.6 preview 2.
